Xamarin.Forms.Core/Routing.cs

```diff
@@ -9,22 +9,32 @@ public static class Routing
 		static int s_routeCount = 0;
 		static Dictionary<string, RouteFactory> s_routes = new Dictionary<string, RouteFactory>();
 
+		// This is a route that's implicitly created from casting something like a ShellContent to a ShellItem
 		internal const string ImplicitPrefix = "IMPL_";
+
+		// These are routes where the user has specified the item in xaml but haven't supplied a route
+		internal const string ImplicitDefaultPrefix = "IMPL_DEFAULT_";
+
 		const string _pathSeparator = "/";
 
 		internal static string GenerateImplicitRoute(string source)
 		{
+			source = source.Replace(ImplicitDefaultPrefix, ImplicitPrefix);
 			if (IsImplicit(source))
 				return source;
 			return String.Concat(ImplicitPrefix, source);
 		}
-		internal static bool IsImplicit(string source)
+
+		internal static bool IsImplicit(string source, bool excludeDefault = false)
 		{
+			if (excludeDefault && source.StartsWith(ImplicitDefaultPrefix, StringComparison.Ordinal))
+				return false;
+
 			return source.StartsWith(ImplicitPrefix, StringComparison.Ordinal);
 		}
-		internal static bool IsImplicit(BindableObject source)
+		internal static bool IsImplicit(BindableObject source, bool excludeDefault = false)
```

Comment:
as we discussed... this would be very helpful to have made public so that Custom Navigation Services can determine if a node is Implicit or not. Also 100% ok with it being hidden from intellisense.